### ISSUE TITLE
Convention based route builder

### ISSUE THREAD

**Issue by stefandevo** [dj-nitehawk/FastEndpoints#29]
I have another suggestion - which I can help you implement.

The idea would be to have an option to activate a convention based route builder.
This means that you don't have to specify your route if you want it to follow the namespace in which you are working.

From your same Web application:

```
namespace Customers.List.Recent;
public class Endpoint : EndpointWithoutRequest
{
    public override void Configure()
    {
        Verbs(Http.GET);
        Routes("/customer/list/recent");
    }
}
```

So basically the `Routes` can be build automatically using the namespace.

You would still be able to set (additional) routes manually if you choose for this convention based option.

In this option you can add "ignore" words (for example, if your namespace would contain MyCompany.Features.Customers.List.Recent you could add "MyCompany" and "Features" as ignore words).
Furthermore you could decide on formatting: lower-case, pascal-wording, etc.
`Inventory.GetProduct` -> `inventory/get-product` for example.

Again, this would be totally optional as all the rest and I don't think it would affect performance when building the routes.

What do you think? Would it be an idea to add (I can make a PR if you want).
If you don't like it, or think the lib we be "too much" just say :-)

**Comment by dj-nitehawk** [comment]
i personally do not like convetion based routing. because i like to see the exact full route/url when i look at the endpoint. so i don't have to spend even a minute amount of brain capacity to figure out what the route is for an endpoint.
also `KISS` is the only design philosophy i try to follow religiously. and i believe this feature would be poking a stick at KISS ;-)

let's see... if there's more interest in this from the community and enough ppl ask for it, then we will reopen this ticket and work on it in the future.

but hey... keep the ideas coming... you have been bloody awesome so far mate!

**Comment by stefandevo** [comment]
Okay understood! No problem. I can easily add it for my own project in base classes. So I am good.

**Comment by stefandevo** [comment]
Hi @dj-nitehawk - sleeping over it :-) You are right, it's not a good idea to have.
I was experimenting with it yesterday, and found out that a small refactoring in your code could break your api endpoints (because they would change automatically). I will implement for myself a SegmentBuilder utility; just want to standardise the routes (so to easily use the feature name, the action, etc) - but that's something that belongs in the projects itself, not in this library. Thanks for your insight.

**Comment by dj-nitehawk** [comment]
yeah i frequently move around and re-organize my vertical slice/features as the app evolves. having to manage/update the JS clients everytime that happens is a pain in the you know what. route urls should be considered a static contract with the outside world.

quick question... how/where did you hear about fastendpoints?

**Comment by stefandevo** [comment]
> quick question... how/where did you hear about fastendpoints?

I was googling about architectures for .net. Been a long time user of servicestack in the past, current environment is clean architecture/onion and just peeking around for alternatives. Then I found the vertical slice architecture, found https://mehdihadeli.github.io/awesome-software-architecture/architectural-style/vertical-slice-architecture/ this site. One of the articles lead to here 👍

**Comment by dj-nitehawk** [comment]
good stuff :-)
